lib/graphql/name_validator.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 module GraphQL
   class NameValidator
-    VALID_NAME_REGEX = /^[_a-zA-Z][_a-zA-Z0-9]*$/
+    VALID_NAME_REGEX = /^[_a-zA-Z][_a-zA-Z0-9]*\??$/
```

Comment:
I'm surprised this was necessary,  I thought it only tested the finally-built GraphQL schema. Since the `?` is removed earlier on, I thought this could remain unchanged. Does it work without changing this Regexp? 

This pattern is used in lots of places in the gem, and I don't think we should add `?` to it. (For example, I think this will allow enum values to contain `?`, which is not valid GraphQL.)